server/src/main/java/org/opensearch/cluster/metadata/MetadataUpdateSettingsService.java

```diff
@@ -104,6 +107,7 @@ public MetadataUpdateSettingsService(
         this.indexScopedSettings = indexScopedSettings;
         this.indicesService = indicesService;
         this.shardLimitValidator = shardLimitValidator;
+        this.awarenessReplicaBalance = new AwarenessReplicaBalance(clusterService.getSettings(), clusterService.getClusterSettings());
```

Comment:
Lets inject a singleton `AwarenessReplicaBalance` in Node.java see `ShardLimitValidator` instead of creating multiple instance, I see them getting created at `MetadataCreateIndexService.java` as well

Reply:
Makes sense, done it . Thanks for pointing this out.